app/src/processing/app/macosx/ThinkDifferent.java

```diff
@@ -45,6 +47,20 @@
 
   static public void init() {
     Application application = Application.getApplication();
+
+    application.addAppEventListener(new AppReOpenedListener() {
+      @Override
+        public void appReOpened(AppReOpenedEvent aroe) {
+          try {
+            if (Base.INSTANCE.getEditors().size() == 0) {
+              Base.INSTANCE.handleNew();
+            }
+          } catch (Exception e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
```

Comment:
Better error handling here?

Reply:
Absolutely (no idea about how to handle it anyway)